src/basic/conf-files.c

```diff
@@ -123,6 +123,27 @@ static int base_cmp(char * const *a, char * const *b) {
         return strcmp(basename(*a), basename(*b));
 }
 
+static int get_sorted_files_from_hashmap(Hashmap *fh, char ***ret) {
+        _cleanup_free_ char **sv = NULL;
+        char **files = NULL;
+
+        assert(ret);
+
+        sv = hashmap_get_strv(fh);
+        if (!sv)
+                return -ENOMEM;
+
+        /* The entries in the array given by hashmap_get_strv() are still owned by the hashmap. */
+        files = strv_copy(sv);
+        if (!files)
+                return -ENOMEM;
+
+        typesafe_qsort(files, strv_length(files), base_cmp);
```

Comment:
Maybe add one more cleanup commit to use `path_extract_filename()` instead of `basename()` in `base_cmp()`?